lib/src/visible_range.dart

```diff
@@ -29,6 +37,9 @@ abstract class VisibleRange {
   const factory VisibleRange.week() = WeekVisibleRange;
 
   final int visibleDays;
+  final int swipeRange;
+
+  LocalDate getPeriodStartDate(LocalDate date, DayOfWeek firstDayOfWeek);
```

Comment:
Isn't/shouldn't this be covered by `getTargetPageForFocus`?

Reply:
period getPeriodStartDate is used to identify initialDate (center date basically), where is `getTargetPageForFocus` now just calculates next page without any knowledge about the center date or anything like that